probe-rs/src/debug/debug_info.rs

```diff
@@ -747,22 +791,93 @@ impl DebugInfo {
             };
 
             // Part 1-b: Check LR values to determine if we can continue unwinding.
-            // TODO: ARM has special ranges of LR addresses to indicate fault conditions. We should check those also.
             if let Some(check_return_address) = unwind_registers.get_return_address() {
                 if check_return_address.is_max_value() || check_return_address.is_zero() {
                     // When we encounter the starting (after reset) return address, we've reached the bottom of the stack, so no more unwinding after this.
                     stack_frames.push(return_frame);
                     tracing::trace!("UNWIND: Stack unwind complete - Reached the 'Reset' value of the LR register.");
                     break;
+                } else {
+                    // Part 1-c: If we just processed an exception handler,  we need to update the `unwind_registers` to match the frame that invoked the exception handler.
+                    if exception_return_context != ExceptionReturnContext::NoException {
+                        tracing::trace!(
+                            "UNWIND: Stack unwind reached an exception handler {exception_return_context:?}",
+                        );
+                        match core.core_type() {
+                            CoreType::Armv6m | CoreType::Armv7em | CoreType::Armv7m => {
+                                // TODO: Currently, this is only tested for ExceptionReturnContext::ToThreadMainStack.
+                                // This appears to cover all my embassy-rs and RTIC usage scenarios, and I have been unable to trigger the other two cases in normal usage.
+                                // We need to create custome test cases and an appropriate implementation for the other two cases
+                                //   i.e., use PSP instead of SP where appropriate)
+                                if matches!(
+                                    exception_return_context,
+                                    ExceptionReturnContext::ToHandler(_)
+                                        | ExceptionReturnContext::ToThreadProcessStack(_)
+                                ) {
+                                    // Report the issue, but complete the unwind.
+                                    tracing::error!(
+                                        "UNWIND: Unwinding to an exception handler is not yet implemented for {exception_return_context:?}. This may result in incorrect stack frames."
+                                    );
+                                }
+                                if let Some(msp_register) =
+                                    unwind_registers.get_register_by_name("SP")
```

Comment:
I would prefer to refer to the register in another way, not with a string. The stack pointer should be accessible in a more "typed" way.